### PR TITLE
LNK-2740: Measure Evaluator Unit Tests

### PR DESCRIPTION
**Pull request by c-schuler**
Work done so far:
- Instantiation tests for MeasureEvaluator
- Evaluation tests for cohort, proportion, ratio and continuous variable measures
- Setup builders for patient data and artifacts

TODO:
- [x]  Harden evaluator logging and validation
- [ ]  Plugin a better caching strategy 
- [ ]  Add tests for:
  - [x]  SDEs
  - [x]  Terminology
  - [ ]  Edge cases

## Summary by CodeRabbit

- **New Features**
  - Introduced robust measure evaluation tests for cohort, proportion, ratio, and continuous variable measures.
  - Added the `PatientDataBuilder` for streamlined creation of FHIR resources in tests.

- **Bug Fixes**
  - Enhanced error handling in `MeasureEvaluator` for unsupported FHIR versions and empty bundle entries.

- **Tests**
  - Introduced comprehensive test cases for `MeasureEvaluator` instantiations and evaluations.
  - Added methods to `PatientDataBuilder` for constructing test data.

- **Chores**
  - Added JUnit Jupiter dependency for enhanced testing capabilities.

### CODE REVIEW

**Comment by seanmcilvenna** [comment]
@coderabbitai review

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/KnowledgeArtifactBuilder.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                
+                define "Denominator":
+                  true
+                  
+                define "Denominator Exclusion"
+                  false""";
+        public static Measure measure() {
+            Measure measure = new Measure();
+            measure.addLibrary(LIBRARY_URL);
+            measure.setScoring(new CodeableConcept().addCoding(new Coding().setCode("proportion")));
```
Comment (seanmcilvenna) [inline Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/KnowledgeArtifactBuilder.java]: We're primarily targeting cohort measures, right now. I like testing additional scoring methods here, but can we make sure that "Cohort" is among the tests, since that's what we're using now?

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]
* Can we get some tests for other classes in the service, like the `MeasureCache` class?
* Would like to see some negative tests, as well (so-to-speak)... which would require only a _slightly_ more complex CQL that does more than `return true`
* One of the things that is done often in our use case is to have definitions that return full resources (which get represented as contained resources on the resulting MeasureReport. Can you include a test that demonstrates that successfully?

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/MeasureEvaluatorInstantiationTests.java
Comment (smailliwcs) [inline Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/MeasureEvaluatorInstantiationTests.java:1]: Agree with the intent behind these test cases, but they should probably be applied to `MeasureDefinitionBundleValidator`, because that's the class we use to validate client-provided bundles.  By the time we get to `MeasureEvaluator` instantiation (which is what we're testing here), it's too late to reject an invalid bundle.

**Review by smailliwcs — CHANGES_REQUESTED** [review]

**Review by seanmcilvenna — APPROVED** [review]

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/MeasureEvaluatorInstantiationTests.java
Comment (c-schuler) [reply Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/MeasureEvaluatorInstantiationTests.java:1]: @smailliwcs I have applied your feedback. Thanks!

**Comment by seanmcilvenna** [comment]
@c-schuler can you make sure this branch gets updated with the latest from `dev`?
@smailliwcs can you give look over this?

**Review by smailliwcs — APPROVED** [review]

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/KnowledgeArtifactBuilder.java (hunk 1 above)
Comment (c-schuler) [reply Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/KnowledgeArtifactBuilder.java]: Added additional cohort measure tests

**Review by sdmcgeown — APPROVED** [review]

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/PatientDataBuilder.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.hl7.fhir.r4.model.*;
+
+public class PatientDataBuilder {
+
+    public static Patient patient() {
+        var patient = new Patient();
+        patient.setId("Patient/simple-patient");
+        return patient;
+    }
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/PatientDataBuilder.java]: Updated

File: Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/PatientDataBuilder.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    public static Encounter encounter() {
+        var encounter = new Encounter();
+        encounter.setId("Encounter/simple-encounter");
+        encounter.setSubject(new Reference("Patient/simple-patient"));
+        encounter.setStatus(Encounter.EncounterStatus.FINISHED);
+        encounter.setClass_(new Coding().setSystem("http://terminology.hl7.org/CodeSystem/v3-ActCode").setCode("AMB"));
+        return encounter;
```
[the remark this answers is not in the document]
Comment (c-schuler) [reply Java/measureeval/src/test/java/com/lantanagroup/link/measureeval/services/PatientDataBuilder.java]: Updated
